### PR TITLE
Easier strict-mode issues

### PR DESCRIPTION
**Pull request by MilesBHuff**
This commit contains fixes for all of the not-virtually-impossible strict-mode settings.

### CODE REVIEW

**Comment by MilesBHuff** [comment]
Hold up -- 8ca7ef698651abce42e8a514e52c2433f5de10ac broke something.

**Comment by MilesBHuff** [comment]
Okay, should be good to merge, now.

File: base/windows/shop/SellRepairMenu.ts [hunk 1]
```diff
@@ -255,7 +255,7 @@ export class SellRepairMenu {
                         this.on_character_select.bind(
                             this,
                             "sell_follow_up",
-                            this.selected_char_index,
+                            // this.selected_char_index,
```
Comment (jjppof) [inline base/windows/shop/SellRepairMenu.ts:258]: If not necessary, we can remove it.

Comment (MilesBHuff) [reply base/windows/shop/SellRepairMenu.ts:258]: I made that commit before your Discord reply -- let me rebase.

Comment (MilesBHuff) [reply base/windows/shop/SellRepairMenu.ts]: Done.

File: base/game_events/GameEvent.ts [hunk 2]
```diff
@@ -112,16 +112,10 @@ export abstract class GameEvent {
...
-        const original_method = descriptor.value;
-        descriptor.value = function (...args) {
-            const data: GoldenSun = this.data;
-            if (data.hero.on_reveal) {
-                (this.data.info.field_abilities_list.reveal as RevealFieldPsynergy).finish(false, false);
-            }
-            return original_method.apply(this, args);
-        };
-        return descriptor;
+    check_reveal() {
```
Comment (jjppof) [inline base/game_events/GameEvent.ts:115]: Please also update the description of this function.

Comment (MilesBHuff) [reply base/game_events/GameEvent.ts:109]: Ah!  Right, my bad.  Should be fixed, now.

**Review by jjppof — DISMISSED** [review]

**Comment by MilesBHuff** [comment]
Ach, Prettier strikes again.  One sec.

**Review by jjppof — APPROVED** [review]

**Pull request merged**